lang/csharp/src/apache/main/Schema/RecordSchema.cs

```diff
@@ -162,13 +281,30 @@ private static Field createField(JToken jfield, int pos, SchemaNames names, stri
             return new Field(schema, name, aliases, pos, doc, defaultValue, sortorder, props);
         }
 
-        private static void addToFieldMap(Dictionary<string, Field> map, string name, Field field)
+        private static void addParsedFieldToFieldMap(Dictionary<string, Field> map, string name, Field field)
         {
             if (map.ContainsKey(name))
                 throw new SchemaParseException("field or alias " + name + " is a duplicate name");
             map.Add(name, field);
         }
 
+        private static void addToFieldMap(Dictionary<string, Field> map, string name, Field field)
```

Comment:
Why is addParsedFieldToFieldMap needed? It seems identical to addToFieldMap. Can we just keep `addToFieldMap` as is ans use it instead of `addParsedFieldToFieldMap`?

Reply:
Is it just a workaround to change the exception type? If yes I would change the exception to the new type and call it out. This seems to me more confusing :)

Reply:
Yes, tried to keep it backward compatible. And it's actually just the inner exception (I can RecordSchema.NewInstance to throw SchemaParseException anyway).  Is that OK to break compatibility of inner exception? 

What do you mean by "call it out"?

Reply:
With the call out I meant to make a release note that `in case of duplicate record field, AVroExc eption, or whatever exceptionn is thrown instead SchemaParseException.

I would have only one `addToFieldMap` method, and change the exception type to something more meaningful if SchemaParseException is not really correct (and create the release note). Or keep the original addToFieldMap as is. I just dont really see theadvantage of having `addParsedFieldToFieldMap`.

Reply:
Done